File: 1680-concatenation-of-consecutive-binary-numbers/1680-concatenation-of-consecutive-binary-numbers.js
/**
 * @param {number} n
 * @return {number}
 */
var concatenatedBinary = function(n) {
  const mod = 10 ** 9 + 7;
  let [res, len] = [1, 0b100];
  
  for (let i = 2; i <= n; i++) {
    if (i === len) len <<= 1;
    res = (res * len + i) % mod;
  }
  
  return res;
};

File: 1680-concatenation-of-consecutive-binary-numbers/1680-concatenation-of-consecutive-binary-numbers.py
class Solution:
    def concatenatedBinary(self, n: int) -> int:
        MOD, res = 10 ** 9 + 7, 1        
        for i in range(2, n + 1):
            res = (res << i.bit_length() | i) % MOD
        return res
